frontend/src/core/l10n/components/AppLocalizationProvider.js

```diff
@@ -29,9 +30,21 @@ type InternalProps = {|
  */
 export class AppLocalizationProviderBase extends React.Component<InternalProps> {
     componentDidMount() {
-        // $FLOW_IGNORE: we count on the 'lang' attribute being set.
-        const locale = document.documentElement.lang;
-        this.props.dispatch(l10n.actions.get([locale]));
+        if (document.documentElement && document.documentElement.lang) {
+            // The user has chosen a specific locale, show that.
+            this.props.dispatch(
+                l10n.actions.get([ document.documentElement.lang ])
```

Comment:
I think this should have `en-US` as the last item, in case the selected locale is not complete. The UI should fall back to English then. You could use `negotiateLanguages` here as well, actually, or just push it if `documentElement.lang !== "en-US"`.

Reply:
That is taken care of in the `get` action, see here: https://github.com/mozilla/pontoon/blob/8d71348cd55f436920e19965e53faa0c08d7af3c/frontend/src/core/l10n/actions.js#L50-L56